profiler/src/ProfilerEngine/Datadog.Profiler.Native/StackSamplerLoop.cpp

```diff
@@ -246,20 +241,13 @@ void StackSamplerLoop::CpuProfilingIteration(void)
                 }
             }
             // don't yield until a thread to sample is found
-
-            // LoopNext() calls AddRef() on the threadInfo before returning it.
-            // This is because it needs to happen under the managedThreads's internal lock
-            // so that a concurrently dying thread cannot delete our threadInfo while we
-            // are just about to start processing it.
-            _targetThread->Release();
-            _targetThread = nullptr;
-
+            _targetThread.reset();
         }
     }
 }
 
 void StackSamplerLoop::CollectOneThreadStackSample(
-    ManagedThreadInfo* pThreadInfo,
+    std::shared_ptr<ManagedThreadInfo>& pThreadInfo,
```

Comment:
I'm not sure we need to pass the smart pointer (especially by ref so not triggering an AddRef?): we could have kept the signature and pass .get() like for CollectStackSample().
Note that the function only uses .get()

Reply:
Later we call a persisting function and we need a copy. So not possible to pass a raw pointer.